girderformindlogger/api/v1/applet.py

```diff
@@ -555,31 +555,46 @@ def setSchedule(self, applet, schedule, **kwargs):
                 "Only coordinators and managers can update applet schedules."
             )
 
+        assigned = {}
         if 'events' in schedule:
-            # need logic for deleting
-            # intersection of of data from the database for the current user and data received from
-            # the front-end
-
             for event in list(schedule['events']):
                 if 'data' in event and 'useNotifications' in event['data'] and event['data']['useNotifications']:
                     if 'notifications' in event['data'] and event['data']['notifications'][0]['start']:
                         # in case of daily/weekly event
-                        exist_notification = PushNotificationModel().findOne(
-                            query={'applet': applet['_id'],
-                                   'creator_id': thisUser['_id']
-                                   })
+                        exist_notification = None
+
+                        if 'id' in event['data']:
+                            assigned[event['data']['id']] = True
 
-                        # should be the logic to update existing event
-                        # PushNotificationModel().update_notification
+                            exist_notification = PushNotificationModel().findOne(
+                                query={'applet': applet['_id'],
+                                    'creator_id': thisUser['_id'],
+                                    'event_id': event['data']['id']
+                                })
 
-                        if not exist_notification:
-                            created_notification = PushNotificationModel().createNotification(
+                        if exist_notification:
+                            PushNotificationModel().replaceNotification(
+                                applet['_id'],
```

Comment:
will be better to add id for response i think
`event['id'] = exist_notification['_id']`